File: SingleNumber.cs
public class Solution {
    public int SingleNumber(int[] nums) {
        
        Array.Sort(nums);
        
        for (int i = 0; i < nums.Length - 1; i+=2)
        {
            if (nums[i + 1] != nums[i])
            {
                return nums[i];
            }
            //Console.WriteLine(nums[i]);
        }
        
        return nums[nums.Length - 1];
    }
}

File: Python/singleNumber.py
class Solution:
    def singleNumber(self, nums: List[int]) -> int:

        nums.sort()
        index = 0

        while index < len(nums) - 1:

            if nums[index] != nums[index + 1]:
                return nums[index]

            index += 2

        return nums[len(nums) - 1]